plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/UriBasedS3SecurityMappingsProvider.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.s3;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import static java.lang.String.format;
+
+public class UriBasedS3SecurityMappingsProvider
+        extends S3SecurityMappingsProvider
+{
+    public UriBasedS3SecurityMappingsProvider(S3SecurityMappingConfig config)
+    {
+        super(config);
+    }
+
+    @Override
+    protected String getRawJSONString()
+    {
+        String urlString = config.getConfigUri().orElseThrow(() -> new IllegalArgumentException("hive.s3.security-mapping.config-uri file is not set"));
+        try {
+            URL url = new URL(urlString);
+            HttpURLConnection con = (HttpURLConnection) url.openConnection();
```

Comment:
We should really use the Airlift HTTP client here, like we do in the rest of Trino, since it integrates with the config system and provides full support for HTTPS configuration rather than relying on JVM defaults. It may feel like overkill here, but provides consistency and makes it more future proof.

You'd do this by changing the bindings in `HiveS3Module` to bind the provider conditionally and inject `S3SecurityMappingsProvider` rather than having `S3SecurityMappingsProviderFactory`. Something like this
```java
private void bindSecurityMapping(Binder binder)
{
    if (buildConfigObject(S3SecurityMappingConfig.class).getConfigFile().isPresent()) {
        binder.bind(S3SecurityMappingsProvider).to(FileBasedS3SecurityMappingsProvider.class).in(Scopes.SINGLETON);
    }
    else if (buildConfigObject(S3SecurityMappingConfig.class).getConfigUri().isPresent()) {
        binder.bind(S3SecurityMappingsProvider).to(UriBasedS3SecurityMappingsProvider.class).in(Scopes.SINGLETON);
        httpClientBinder(binder).bindHttpClient("s3SecurityMapping", ForS3SecurityMapping.class)
                .withConfigDefaults(config -> config
                        .setRequestTimeout(10, SECONDS)
                        .setSelectorCount(1)
                        .setMinThreads(1);
                });
    }
    else {
        return;
    }

    newSetBinder(binder, DynamicConfigurationProvider.class).addBinding()
            .to(S3SecurityMappingConfigurationProvider.class).in(Scopes.SINGLETON);

    checkArgument(!buildConfigObject(HiveConfig.class).isS3SelectPushdownEnabled(), "S3 security mapping is not compatible with S3 Select pushdown");
    checkArgument(!buildConfigObject(RubixEnabledConfig.class).isCacheEnabled(), "S3 security mapping is not compatible with Hive caching");
}
```
See `io.trino.server.security.jwt.JwkService` for a simple example of using the client. You'll need to add an annotation `ForS3SecurityMapping` to qualify the client for injecting (just copy `ForJwk`). Using `createStringResponseHandler()` is probably good for this, just like Jwks.